src/main/kotlin/br/unipar/plano/domain/contratos/model/Contrato.kt

```diff
@@ -0,0 +1,70 @@
+package br.unipar.plano.domain.contratos.model
+
+import br.unipar.plano.domain.pessoas.model.Pessoa
+import br.unipar.plano.domain.planos.model.Plano
+import java.time.LocalDate
+import javax.persistence.*
+
+enum class StatusContrato {
+    ATIVO, CANCELADO
+}
+
+@Entity
+class Contrato(
+
+    @field:EmbeddedId
+    val id: IdContrato,
+
+    @Column(nullable = false)
+    val dataContratacao: LocalDate,
+
+    @Column(nullable = false)
+    val dataContratoFinal: LocalDate,
+
+    @Column(nullable = false)
+    val dataCancelamento: LocalDate?,
+
+    @ManyToOne
+    val plano: Plano,
+
+    @OneToOne
+    val titular: Pessoa,
+
+    @Enumerated(EnumType.STRING)
+    val status: StatusContrato = StatusContrato.ATIVO
+) {
+    fun with(
+        idContrato: IdContrato = this.id,
+        dataContratoFinal: LocalDate = this.dataContratoFinal,
+        plano: Plano = this.plano

```

Comment:
Melhor alterar esse ponto para não permitir a alteração do plano.

```suggestion
        plano: Plano = this.plano

```

Reply:
Feito de forma que só permita alteração de Update de plano